dace/codegen/targets/intel_fpga.py

```diff
@@ -177,6 +181,31 @@ def get_generated_codeobjects(self):
 
         return [host_code_obj] + kernel_code_objs + other_code_objs
 
+    def set_mangled_channel_name(self, var_name, kernel_id):
+        '''
+        Memorize and returns  the mangled name of a global channel
+        The dictionary is organized as (var_name) : {kernel_id: mangled_name)
+        '''
+
+        if kernel_id not in self.channel_mangle[var_name]:
+            existing_count = len(self.channel_mangle[var_name])
+            suffix = f"_{existing_count}" if existing_count > 0 else ""
+            mangled_name = f"{var_name}{suffix}"
+            self.channel_mangle[var_name][kernel_id] = mangled_name
+        return self.channel_mangle[var_name][kernel_id]
+
+    def get_mangled_channel_name(self, var_name, kernel_id):
```

Comment:
This naming confuses me a bit, since both `set` and `get` return the variable name. Why can't they be combined? Maybe one should be implemented in terms of the other?

Reply:
Changing the name would be ok? Like `set_mangled_name` -> `create_mangled_name`
The `set` actually creates the mangled name (if not already present) and returns it .
The get only returns it, it does not have to create it.
They are called in different situations

Reply:
Create is better, but it's still weird, since it's not _necessarily_ creating anything. What is the reason that they cannot be combined in a single function?

Reply:
Because:
- I don't want to create a new mangle if that does not exist
- I want to deal with the "alias", that when we are trying to resolve a channel name that was defined by a `#define`

Reply:
Ok, doesn't the check know if a variable is an alias or not? Anyway, it's not that important, I just didn't really understand the difference between the two functions. I like `create` and `get` more than `set` and `get`.